### DISCUSSION TITLE
Customize Android setup - Pull down to refresh

### DISCUSSION THREAD

**Discussion by BSteffaniak** [tauri-apps/tauri#9037] [Q&A]
I've been getting my tauri app set up with Android and have it successfully working.

However, there are a few questions I have regarding the setup:

1. I am assuming it is alright to modify the contents of src-tauri/gen/android as long as it is not a file that is generated on build (i.e. the files with something like 'THIS IS AN AUTOGENERATED FILE' at the top) -- as well as check them into source control. Is this a correct assumption?
2. Is there an ability to customize the layout of the app by, for example, wrapping the webview in a SwipeRefreshLayout (similar to what is shown [here](https://stackoverflow.com/a/37775633/1305997))?

It seems like the WryActivity owns the webview and doesn't expose it in any way that I am aware of.

The main reason I want to do this is to add a 'pull down to refresh the page' functionality to the app. However, if there is a different way to do that, I am open to ideas.

FYI, I am not very experienced with android development by any means, so please forgive my ignorance.

Thanks!

**Comment by FabianLars** [comment]
1. Yes
2. Hmm, you could try to edit WryActivity in the generated folder and see if tauri overwrites it. If it does, please open an issue here https://github.com/tauri-apps/wry/ as i'm not sure how we should handle this. A specific option just for this? Something similar to tauri's installer templates? 🤷
